GameWindow.java

```diff
@@ -442,216 +459,174 @@ public void run() { // it is necessary to check if the program is 'running' or n
+			
+			
+			
+		if(state == STATE.PLAY)
+		{
+			
+			
 			// FOX WAVEDASH
-			if(HardMode.isSelected() && FoxWavedash.isSelected()) {
-				YnoteFW.setFallSpeed(5.5);
-				RnoteFW.setFallSpeed(5.5);
-				stickDLeftnoteFW.setFallSpeed(5.5);
-			}
-			if(MediumMode.isSelected() && FoxWavedash.isSelected()) {
-				YnoteFW.setFallSpeed(2.75);
-				RnoteFW.setFallSpeed(2.75);
-				stickDLeftnoteFW.setFallSpeed(2.75);
-			}
-			if(EasyMode.isSelected() && FoxWavedash.isSelected()) {
-				YnoteFW.setFallSpeed(1.375);
-				RnoteFW.setFallSpeed(1.375);
-				stickDLeftnoteFW.setFallSpeed(1.375);
-			}
-			if(HardMode.isSelected() && multi.isSelected()) {
-				XnoteMS.setFallSpeed(7);
-				BnoteMS.setFallSpeed(7);
-				stickDownnoteMS.setFallSpeed(7);
-				stickDownnoteMS2.setFallSpeed(7);
-				stickDownnoteMS3.setFallSpeed(7);
-			}
-			if(MediumMode.isSelected() && multi.isSelected()) {
-				XnoteMS.setFallSpeed(3.5);
-				BnoteMS.setFallSpeed(3.5);
-				stickDownnoteMS.setFallSpeed(3.5);
-				stickDownnoteMS2.setFallSpeed(3.5);
-				stickDownnoteMS3.setFallSpeed(3.5);
-			}
-			if(EasyMode.isSelected() && multi.isSelected()) {
-				XnoteMS.setFallSpeed(1.75);
-				BnoteMS.setFallSpeed(1.75);
-				stickDownnoteMS.setFallSpeed(1.75);
-				stickDownnoteMS2.setFallSpeed(1.75);
-				stickDownnoteMS3.setFallSpeed(1.75);
-			}
-			if (FoxWavedash.isSelected()) {
-               
-				// sets highscore for fox wavedash
-				highscore = hs.hScore(0);
-                System.out.println("Y: "+YnoteFW.getY());
-				delayFalling();
-
-				YnoteFW.tick();
-
-				RnoteFW.tick();
-
-				stickDLeftnoteFW.tick();
-				
-				
-				if (Physics.Collision(YnoteFW, ybutton2) && YnoteFW.getY() > 352 && YnoteFW.getY() < 360 && ybutton2.getX() == 339) {
-					a = 1;
-					YnoteFW.setY(-54);
-					PlaySound(cSound);
-					YnoteFW.setFalling(false);
-
-				}
-
-				if (Physics.Collision(RnoteFW, rbutton2) && RnoteFW.getY() > 335 && RnoteFW.getY() < 343 && rbutton2.getX() == 348) {
-					b = 1;
-
-					RnoteFW.setY(-74);
-					PlaySound(aSound);
-					RnoteFW.setFalling(false);
-				}
 
-				if (Physics.Collision(stickDLeftnoteFW, stickDLeft2) && stickDLeftnoteFW.getY() > 364 && stickDLeftnoteFW.getY() < 371
-						&& stickDLeft2.getX() == 264 && stickDLeft2.getZ() == 8) {
-					c = 1;
-					stickDLeftnoteFW.setY(-76);
+			
+		if (FoxWavedash.isSelected()) {
+			score = advancedScores[0];
+			
+			//sets highscore for fox wavedash
+			
+			highscore = hs.hScore(0);
+			
+			delayFalling();
 
-					// System.out.println(score);
-					stickDLeftnoteFW.setFalling(false);
+			YnoteFW.tick();
 
-				}
+			RnoteFW.tick();
 
-				//System.out.println("counter: " + counter);
+			stickDLeftnoteFW.tick();
 
-				if (counter == 2)
-				{
-					
-					state = STATE.PRESENTSCORE;
-				}
+			
+			if (Physics.Collision(YnoteFW, ybutton2) && YnoteFW.getY() > 353 && ybutton2.getX() == 319) {
+				a = 1;
+				YnoteFW.setY(-54);
+				PlaySound(cSound);
+				YnoteFW.setFalling(false);

```

Comment:
It might be good to add a `reset()` function to `Note` which is able to reset its Y position and its `setFalling` value. That way you don't have to hardcode the value again here, which leaves room for mistakes.

Also, if you include the two "magic numbers" (`353` and `319`) in the Note class as more parameters, then we can factor this out of the technique using the same looping idea earlier.

Finally, if you make sure the map key for the `sounds` map matches the `getName()` (or whatever you want to call it) of the `Note` in question, then you can just `PlaySound(sounds.get(n.getName()))`--where `n` is the note in question--which is also a necessary step towards being able to get rid of all of the special variables here.